build.gradle

```diff
@@ -685,7 +698,7 @@ task requireJavadoc(type: JavaExec, group: 'Documentation') {
     description = 'Ensures that Javadoc documentation exists in source code.'
     mainClass = 'org.plumelib.javadoc.RequireJavadoc'
     classpath = configurations.requireJavadoc
-    args 'checker/src/main/java', 'checker-qual/src/main/java', 'checker-util/src/main/java', 'dataflow/src/main/java', 'framework/src/main/java', 'framework-test/src/main/java', 'javacutil/src/main/java'
+    args javadocDirs.collect { it.srcDirs*.absolutePath }.flatten()
```

Comment:
Can you add a comment what the `srcDirs*` does?

Reply:
* is This is Groovy's spread-dot operator for iterating a collection. https://groovy-lang.org/operators.html#_spread_operator

Reply:
@wmdietl No, I get the warning when define `javadocDirs`. It will first evaluate root project gradle file and did not find the definition of testannotation sourceset because the subproject's haven't been evaluated yet.

Reply:
Can you try changing the evaluation order? It seems wrong to define the `testannotations` property in the root project where it isn't used. It should only be defined for the sub-projects that actually define it.

Reply:
Added in new commit.

Reply:
Nice solution!